apps/src/lib/util/firehose.js

```diff
@@ -0,0 +1,114 @@
+ *       user_id: current_user.id,           // OPTIONAL
+ *       script_id: script.id,               // OPTIONAL
+ *       level_id: level.id,                 // OPTIONAL
+ *       project_id: project.id              // OPTIONAL
+ *       event: 'drowning',                  // REQUIRED
+ *       data_int: 2                         // OPTIONAL
+ *       data_float: 0.31                    // OPTIONAL
+ *       data_string: 'code.org rocks'       // OPTIONAL
+ *       data_json: JSON.stringify(x)        // OPTIONAL
+ *     }
+ *   );
+ * Usage:
+ *   FirehoseClient.putRecordBatch(
+ *     'analysis-events',
+ *     [
+ *       {
+ *         created_at: DateTime.now,           // REQUIRED
+ *         environment: 'production',          // REQUIRED
+ *         study: 'underwater basket weaving', // REQUIRED
+ *         study_group: 'control',             // OPTIONAL
+ *         user_id: current_user.id,           // OPTIONAL
+ *         script_id: script.id,               // OPTIONAL
+ *         level_id: level.id,                 // OPTIONAL
+ *         project_id: project.id              // OPTIONAL
+ *         event: 'drowning',                  // REQUIRED
+ *         data_int: 2                         // OPTIONAL
+ *         data_float: 0.31                    // OPTIONAL
+ *         data_string: 'code.org rocks'       // OPTIONAL
+ *         data_json: JSON.stringify(x)        // OPTIONAL
+ *       },
+ *       {
+ *         created_at: DateTime.now,           // REQUIRED
+ *         environment: 'production',          // REQUIRED
+ *         study: 'underwater basket weaving', // REQUIRED
+ *         study_group: 'control',             // OPTIONAL
+ *         user_id: current_user.id,           // OPTIONAL
+ *         script_id: script.id,               // OPTIONAL
+ *         level_id: level.id,                 // OPTIONAL
+ *         project_id: project.id              // OPTIONAL
+ *         event: 'drowning',                  // REQUIRED
+ *         data_int: 2                         // OPTIONAL
+ *         data_float: 0.31                    // OPTIONAL
+ *         data_string: 'code.org rocks'       // OPTIONAL
+ *         data_json: JSON.stringify(x)        // OPTIONAL
+ *       },
+ *     ]
+ *   );
+ */
+// TODO(asher): Add the ability to queue records individually, to be submitted
+// as a batch.
+export default class FirehoseClient {
+  /**
+   * Pushes one data record into the delivery stream.
+   * @param {string} deliveryStreamName The name of the delivery stream.
+   * @param {hash} data The data to push.
+   */
+  putRecord(deliveryStreamName, data) {
+    FIREHOSE.putRecord(
+      {
+        DeliveryStreamName: deliveryStreamName,
+        Record: JSON.stringify(data),
+      },
+      function (err, data) {
+        // TODO(asher): This is here to assist debugging and should be removed
+        // when it is no longer useful.
+        console.error("Error pushing event data" + err);
```

Comment:
Note, this callback will trigger asynchronously on either success or failure (similarly, below), so you should check for 'err' to determine whether the call succeeded or not.